### PR TITLE
Implement sentinel errors as real constants

### PR DESCRIPTION
**Pull request by ansiwen**
### What does this PR achieve? Why do we need it?

The errors created with errors.New() can't be used in constant
values.  Hence the sentinel errors could be overwritten by accident.
However, pure strings can be stored as constant values.  This change
replaces the sentinel errors with an error type that is isomorph to a
pure string and consequently defines them as constant values.

Based on [this article](https://dave.cheney.net/2016/04/07/constant-errors).

Mutually exclusive with #1380 

Signed-off-by: Sven Anderson <sven@redhat.com>

### CODE REVIEW

**Comment by phlogistonjohn** [comment]
I'm struggling to see what issues this PR solves. Overwriting the errors seems fairly unlikely to me.

**Comment by ansiwen** [comment]
This is just spin-off code that I found from the times when I wrote the other error PRs and I thought I share it instead of throwing it away. It's certainly low prio. On the other hand, does it hurt? If we will go with the solution from #1380 you can just ignore this of course.

**Comment by phlogistonjohn** [comment]
After giving this some more thought and re-reading the associated blog post I'm comfortable with this in the abstract. Here's what I'd like to see happen though: let's make the "ConstError" a new sub-package under pkg so that other parts of heketi (I'm specifically thinking of executors) can use this new type.
Then we can document the new type with some doc comments and a in-code link to the blog post. Having done that I'd think I'd be comfortable merging it.

**Comment by phlogistonjohn** [comment]
@ansiwen ping. Please see me last comment.

**Comment by ansiwen** [comment]
@phlogistonjohn sure, we can do that. but before putting effort into that, we should first decide, if we don't prefer replacing them with something that has back-traces to the point of where the error happened, like I proposed in #1380. I would prefer the latter, because it proved to be very helpful for me during debugging before. This PR is like a second choice for me, if we decide to not use the back-traceable errors package.

**Comment by phlogistonjohn** [comment]
I honestly don't see the two as mutually exclusive, especially if we put the new type in it's own package and can thus the implementation can be changed more independently.

**Comment by ansiwen** [comment]
I think I'm confused.

#1380 would replace every line of this PR, so how is it _not_ mutually exclusive? Maybe I don't understand correctly what you mean. So let me rephrase:

You are proposing to put the
```
type constErr string

func (e constErr) Error() string {
	return string(e)
}
```
into a file in pkg, but the actual instances of the errors from this PR still stay in the glusterfs/errors.go, right? (You couldn't put more into the pkg, like a New() function or so, since that doesn't work with constants. It would have to be created directly casting constant string into that type, like in this PR.)

So we would maybe later replace that with the errtag Instances, making the consterr pkg unused, no?

Or do you maybe suggest to implement the errtag in a way that they wrap arbitrary other errors, and we could use the consterr objects there?

**Comment by phlogistonjohn** [comment]
Thank you for your contribution however I think we'll decline taking this approach. I learned a lot from this PR but I think it's a bit too much change for such a fundamental part of heketi. If we were to fundamentally change our error handling I'd prefer we look into the new native approach in Go 1.13.
